buildSrc/src/main/kotlin/global-compiler-options.gradle.kts

```diff
@@ -46,3 +55,9 @@ tasks.withType<Kotlin2JsCompile>().configureEach {
 tasks.withType<KotlinNativeCompile>().configureEach {
     compilerOptions { freeCompilerArgs.add("-Xpartial-linkage-loglevel=ERROR") }
 }
+
+tasks.withType<KotlinCompilationTask<*>>().configureEach {
+    doFirst {
```

Comment:
I suggest printing only the additional args if they were provided: `if (kotlin_additional_cli_options?.isNotEmpty()) logger.info(...)`

Reply:
I think it's useful to have all options in log. now it look like this: 
Added Kotlin compiler flags: -Xexplicit-api=strict, -Xsuppress-version-warnings, -Xexpect-actual-classes, -P, plugin:org.jetbrains.kotlinx.serialization:disableIntrinsic=false, -Xreport-all-warnings, -Xrender-internal-diagnostic-names, -nowarn